.circleci/generate_config.py

```diff
@@ -0,0 +1,264 @@
+)
+parser.add_argument(
+    "--pr-branch", help="file containing the test definitions", type=str
+)
+
+args = parser.parse_args()
+
+
+def generate_workflow(config):
+    if args.workflow == "generate":
+        workflow_generate(config)
+    
+    if args.workflow.startswith("generate-"):
+        workflow_generate_scheduled(config)
+
+    if args.workflow == "commit-generated":
+        workflow_commit_generated_download_data(config)
+
+    return config
+
+
+## WORKFLOWS
+
+def workflow_generate(config):
+    config = workflow_generate_launch_command(config)
+    config = workflow_generate_store_artifacts_command(config)
+
+    jobs = config["workflows"]["generate"]["jobs"]
+
+    generateRequires = []
+
+    for i in range(len(versions)):
+        version = versions[i]["name"]
+        branch = args.arangodb_branches[i]
+        if branch == "undefined":
+            continue
+
+        print(f"Creating compile job for version {version} branch {branch}")
+
+        openssl = "3.0.9"
+        if not "enterprise-preview" in branch:
+            openssl = findOpensslVersion(branch)
+            print(f"found OpenSSL Version {openssl}")
+
+        compileJob = {
+            "compile-linux": {
+                "context": ["sccache-aws-bucket"],
+                "name": f"compile-{version}",
+                "arangodb-branch": branch,
+                "version": version,
+                "openssl": openssl,
+                "requires": ["approve-workflow"]
+            }
+        }
+        generateRequires.append(f"compile-{version}")
+        jobs.append(compileJob)
+
+    generateJob = {
+        "build-with-generated": {
+            "name": args.workflow,
+            "generators": "<< pipeline.parameters.generators >>",
+            "commit-generated": "<< pipeline.parameters.commit-generated >>",
+            "create-pr": "<< pipeline.parameters.create-pr >>",
+            "pr-branch": "<< pipeline.parameters.pr-branch >>",
+            "requires": generateRequires
+        }
+    }
+
+    deployJob = {
+        "deploy": {
+            "requires": [args.workflow]
+        }
+    }
+    jobs.append(generateJob)
+    jobs.append(deployJob)
+
+    return config
+
+
+def workflow_generate_scheduled(config):
+    config = workflow_generate_launch_command(config)
+    config = workflow_generate_store_artifacts_command(config)
+
+    config["workflows"]["generate"]["jobs"] = []
+    jobs = config["workflows"]["generate"]["jobs"]
+
+    generateRequires = []
+
+    for i in range(len(versions)):
+        version = versions[i]["name"]
+        
+        compileJob = {
+            "compile-linux": {
+                "context": ["sccache-aws-bucket"],
+                "name": f"compile-{version}",
+                "arangodb-branch": f"arangodb/enterprise-preview:{version}-nightly" if versions[i]["alias"] != "devel" else "arangodb/enterprise-preview:devel-nightly",
+                "version": version,
+                "openssl": "3.0.9",
```

Comment:
Is this only a fallback?

Reply:
The openssl in this case just serve as a default value for circleci to be able to download the executor docker image (the arangodb-build docker image) to start the compile job.